### PR TITLE
Added deployment link for milestone 1 

### LINKED ISSUE #29: Milestone 1 Review
- [x] Pull down code 
- [x] run it locally 
- [x] create bug tickets
- [x] place bug tickets in the milestone
- [x] deploy to firebase

### PR DESCRIPTION
**Pull request by Jonathon22**
https://freelancers-7ae52.web.app

## Description
created the navbar, user auth, deployed site

## Related Issue
closes #29
## Motivation and Context
This closes milestone 1 adding all the setup features to start on CRUD. 

## How Can This Be Tested?
git fetch userauth
on the site press the login and logout button. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: src/javascripts/components/navbar/navbar.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import firebase from 'firebase/app';
+import 'firebase/auth';
+
+const logoutEvent = () => {
+  $('#navbar-logout-button').on('click', (e) => {
+    e.preventDefault();
+    // NOTE FOR STUDENTS
+    // Remove session storage if they log out in the same session and in case another user logs in, we want the API check to happen.
+    window.sessionStorage.removeItem('ua');
```
Comment (drteresavasquez) [inline src/javascripts/components/navbar/navbar.js:9]: What does `ua` mean to your team? Do you know what this line is doing?

File: src/javascripts/helpers/data/authData.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import viewHelper from '../viewHelper';
+
+const checkLoginStatus = () => {
+  firebase.auth().onAuthStateChanged((user) => {
+    if (user) {
+      const currentUser = userData.setCurrentUser(user);
+      navbar.navbar(currentUser.name);
+      viewHelper.viewListener('souvenirs-link');
+      $('#google-auth').addClass('hide');
+      $('#welcome-guest').addClass('hide');
```
Comment (drteresavasquez) [inline src/javascripts/helpers/data/authData.js:15]: Love this!

File: src/javascripts/helpers/data/userData.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
+import axios from 'axios';
+import apiKeys from '../../../../apiKeys.json';
+
+const baseUrl = apiKeys.firebaseKeys.databaseURL;
+
+const getUser = () => new Promise((resolve, reject) => {
+  axios
```
Comment (drteresavasquez) [inline src/javascripts/helpers/data/userData.js:7]: Is This getting all the users? The name of the function is get user. If you’re only getting one user that name works, but if you’re getting more than one user it does not. Depending on what you’re trying to do, you may need to rename this function.

File: src/javascripts/helpers/data/userData.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        axios
+          .post(`${baseUrl}/users.json`, user)
+          .then((response) => {
+            const update = { firebaseKey: response.data.name };
+            axios.patch(`${baseUrl}/users/${response.data.name}.json`, update);
+          })
+          .catch((error) => console.warn(error));
+      } else {
+        console.warn('User Already Exists');
+      }
```
Comment (drteresavasquez) [inline src/javascripts/helpers/data/userData.js:36]: Again the ua question 😊

**Review by drteresavasquez — CHANGES_REQUESTED** [review]
A couple of small changes and questions.

File: src/javascripts/components/navbar/navbar.js (hunk 1 above)
Comment (ryanmcnair) [reply src/javascripts/components/navbar/navbar.js:9]: I understand this as User Authentication. The line in question is removing the session storage so when the user logs back in they will start a new session.

File: src/javascripts/helpers/data/authData.js (hunk 2 above)
Comment (ryanmcnair) [reply src/javascripts/helpers/data/authData.js:15]: 🔥

File: src/javascripts/helpers/data/userData.js (hunk 3 above)
Comment (ryanmcnair) [reply src/javascripts/helpers/data/userData.js:7]: it is. I will update the name. After reviewing, I’m going to go back through and add comments so I don’t look like that clown meme.

File: src/javascripts/helpers/data/userData.js (hunk 4 above)
Comment (ryanmcnair) [reply src/javascripts/helpers/data/userData.js:36]: in this case, we are checking to see if the user’s uid is already held in fb, if it’s not there we are adding the uid/displayname/email to fb but if it is there then we get a console.warn

File: src/javascripts/components/navbar/navbar.js (hunk 1 above)
Comment (ryanmcnair) [reply src/javascripts/components/navbar/navbar.js:9]: This is so the API is not hit as many times

**Review by drteresavasquez — APPROVED** [review]
Looks good!

**Review by jacobpaine — APPROVED** [review]
looks good.

**Pull request merged**